### PR TITLE
Rename teacher app sections

### PR DESCRIPTION
**Pull request by clareconstantine**
# Description

Rename teacher application sections so the section numbers reflect the actual order in which they appear in the application, following up to [this change](https://github.com/code-dot-org/code-dot-org/pull/27011) made last year that flipped the Teaching Background and Choose Your Program sections.

[Jira task](https://codedotorg.atlassian.net/browse/PLC-498?atlOrigin=eyJpIjoiNmY3Njg0YmE1ZDZlNDFlZDk3NmJmOTUxZGFkYmU2NzMiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by breville — APPROVED** [review]
Thanks for doing this!  

A lesson I learned when doing Hour of Code tutorials long ago was that if you have a lot of sections/levels that have both a name and an index in the name, get rid of the index and just refer to them by name, and then you never have to worry about changing the order.

That aside, this fix feels appropriate and good!

**Comment by clareconstantine** [comment]
> A lesson I learned when doing Hour of Code tutorials long ago was that if you have a lot of sections/levels that have both a name and an index in the name, get rid of the index and just refer to them by name, and then you never have to worry about changing the order.

I was considering that - do you think I should remove the indices? they are helpful until we need to change them :P

**Comment by breville** [comment]
@clareconstantine If you were already considering it, then it sounds like it might be worth doing.  It's not egregious in this case, but as a matter of principle I think it's a good one :)

**Comment by clareconstantine** [comment]
> @clareconstantine If you were already considering it, then it sounds like it might be worth doing. It's not egregious in this case, but as a matter of principle I think it's a good one :)

@breville I took out the indexes entirely!

File: lib/cdo/shared_constants/pd/teacher2021_application_constants.rb [hunk 1]
```diff
@@ -49,7 +49,11 @@ def self.clean_multiline(string)
           how_heard: 'How did you hear about this program?'
         }
       ),
-      section_2_choose_your_program: {
+      teaching_background:
+        BASE_PAGE_LABELS[:section_2_your_school].slice(
```
Comment (breville) [inline lib/cdo/shared_constants/pd/teacher2021_application_constants.rb:53]: Ah, so we have some mentions of section_X further upstream?  It's fine to keep this change constrained but also interested to ponder how far we can take the numberless principle :)

File: lib/cdo/shared_constants/pd/teacher2021_application_constants.rb [hunk 2]
```diff
@@ -8,16 +8,16 @@ def self.clean_multiline(string)
...
-      section_1_about_you: 'About You',
-      section_2_choose_your_program: 'Choose Your Program',
-      section_3_teaching_background: 'Teaching Background',
-      section_4_professional_learning_program_requirements: 'Professional Learning Program Requirements',
-      section_5_additional_demographic_information: 'Additional Demographic Information and submission',
+      about_you: 'About You',
+      teaching_background: 'Teaching Background',
+      choose_your_program: 'Choose Your Program',
+      professional_learning_program_requirements: 'Professional Learning Program Requirements',
+      additional_demographic_information: 'Additional Demographic Information and submission',
```
Comment (breville) [inline lib/cdo/shared_constants/pd/teacher2021_application_constants.rb:15]: This feels good.. the ordering feels less hardcoded now, and more flexible to change order if we discover that we should!

**Review by breville — APPROVED** [review]

File: lib/cdo/shared_constants/pd/teacher2021_application_constants.rb (hunk 1 above)
Comment (clareconstantine) [reply lib/cdo/shared_constants/pd/teacher2021_application_constants.rb:53]: I agree! I'm going to fix that as part of an upcoming change to simplify how our constants files are structured.

**Pull request merged**